httpx/router.go

```diff
@@ -0,0 +1,89 @@
+package httpx
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
+
+	"github.com/zenledger-io/go-utils/httpx/middlewarex"
+	"github.com/zenledger-io/go-utils/version"
+)
+
+var (
+	defaultRequestTimeout = 60 * time.Second
+)
+
+// APIRouterConfig is configuration for an APIRouter.
+type APIRouterConfig struct {
+	// APIVersion is the URL-based API version. All routes created
+	// on a router with this config will be nested under
+	// "/v1" when APIVersion is 1.
+	APIVersion int
+
+	// APIRouter is the router to be mounted under the API verison.
+	APIRouter chi.Router
```

Comment:
Could we abstract this to not expose the underlying type?

Reply:
I have concerns with the two ways I could imagine us doing this:
1. We design a generic interface that best expresses what an HTTP router ought to do.
2. We copy `chi.Router` and rename it to `httpx.Router` and wrap `chi.Router` with it.

Option 1 is hard to do right and we would have to have a design in place that was better than what `chi.Router` already exposes. I don't have a better design in mind.

Option 2 doesn't make sense because we gain nothing except a layer of abstraction (which needs to be maintained and documented). And, in the event that we change routers we would end up needing to map our lifted `httpx.Router` to whatever other router we chose and end up with another 3rd party router that _looks_ like a `chi.Router` but not something you can google to figure out how to use.

Reply:
Another options would be to expose what we need as we need it but I'm good with it like this too